macros/multiple_databases/to_date.sql

```diff
@@ -2,7 +2,11 @@
 
 {# Cast to date when the input is in a date or a datetime format. This is default behavior for SQL Server. #}
 {%- if target.type == 'databricks' -%}
-    to_date({{ field }}, '{{ var("date_format", "yyyy-MM-dd") }}')
+    case
+        when to_date({{ field }}, '{{ var("date_format", "yyyy-MM-dd") }}') is null
```

Comment:
For which case do we need this condition? In most cases (combined with legacy setting) the parsing will work. When parsing will fail (and the result will be null), the to_date function will also fail, but I might miss a usecase?

Reply:
This is done in line with Snowflake, the "extra" use case is to try parsing the input string to a date without giving an explicit date format, which is optional. By chance, the default is for now the same, but say that this changes in an UDF, then we would get an extra fallback.
I do agree that for now it doesn't bring much added value.